MetadataExtractor/Formats/Eps/EpsReader.cs

```diff
@@ -0,0 +1,420 @@
+                        {
+                            directory.AddError("Unable to process TIFF data: " + ex.Message);
+                        }
+                    }
+                    else if (wmfSize != 0)
+                    {
+                        directory.Set(EpsDirectory.TagWmfPreviewSize, wmfSize);
+                        directory.Set(EpsDirectory.TagWmfPreviewOffset, wmfOffset);
+                    }
+
+                    // TODO avoid allocating byte array here -- read directly from InputStream
+                    Extract(directory, epsDirectories, new SequentialByteArrayReader(reader.GetBytes(postScriptOffset, postScriptLength)));
+                    break;
+                case 0x25215053:
+                    inputStream.Position = startingPosition;
+                    Extract(directory, epsDirectories, new SequentialStreamReader(inputStream));
+                    break;
+                default:
+                    directory.AddError("File type not supported.");
+                    break;
+            }
+
+            return epsDirectories;
+        }
+
+        /// <summary>
+        /// Main method that parses all comments and then distributes data extraction among other methods that parse the
+        /// rest of file and store encountered data in metadata(if there exists an entry in EpsDirectory
+        /// for the found data).  Reads until a begin data/binary comment is found or reader's estimated
+        /// available data has run out (or AI09 End Private Data).  Will extract data from normal EPS comments, Photoshop, ICC, and XMP.
+        /// </summary>
+        /// <param name="directory"></param>
+        /// <param name="directories">list to add directory to and extracted data</param>
+        /// <param name="reader"></param>
+        private void Extract([NotNull] EpsDirectory directory, List<Directory> directories, [NotNull] SequentialReader reader)
+        {
+            var line = new StringBuilder();
+
+            while (true)
+            {
+                line.Length = 0;
+
+                // Read the next line, excluding any trailing newline character
+                // Note that for Windows-style line endings ("\r\n") the outer loop will be run a second time with an empty
+                // string, which is fine.
+                while (true)
+                {
+                    char c = (char)reader.GetByte();
+                    if (c == '\r' || c == '\n')
+                        break;
+                    line.Append(c);
+                }
+
+                // Stop when we hit a line that is not a comment
+                if (line.Length != 0 && line[0] != '%')
+                    break;
+
+                string name;
+
+                // ':' signifies there is an associated keyword (should be put in directory)
+                // otherwise, the name could be a marker
+                int colonIndex = line.IndexOf(":");
```

Comment:
Is this the only use of `IndexOf`? If so, you can make the implementation simpler by searching for a char rather than a string.

Reply:
Most of these are addressed. Let me know if there are any other issues. Thanks